eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Rules/AdaPreservation.hs

```diff
@@ -0,0 +1,656 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Test.Cardano.Ledger.Shelley.Rules.AdaPreservation (
```

Comment:
Thanks for the suggestion, I separated the renames in a different commit (after using `git mv` as suggested) for the three files that I have renamed  (Pool, PoolReap and Deleg). 

However, not sure about this technique for the files that I created by extracting content from the big original file, `TestChain` (including this file AdaPreservation). If i do `git mv` it will rename the whole file, no?  
Is there a way to git mv just a part of a file? Or how can  i keep the history for parts of a file?

Reply:
Very good question. This can be done with two separate commits:

```shell
$ cp TestChain.hs TestChainBak.hs
$ git add TestChainBak.hs
$ git mv AdaPreservation.hs
$ git commit TestChainBak.hs AdaPreservation.hs -m "Make a copy of two files with the same history"
$ git mv TestChainBak.hs TestChain.hs
$ git commit TestChain.hs -m "Moved back into the original"
```

At this point you have two files with the desired names and the same git history. Subsequent commit can include modification to both of the files: 
* `TestChain.hs` has stuff removed that now will live in `AdaPreservation.hs`
* `AdaPreservation.hs` has the stuff removed that stayed in `TestChain.hs`

This is were I stole it from: https://stackoverflow.com/questions/3887736/keep-git-history-when-splitting-a-file#answer-33992444

I know it is a bit elaborate way of going about it., but it is worth it.

Reply:
That's crazy. I love it

Reply:
Done, but at the cost of many extra commits (2  extra for each extraction, so 8 extra commits). 
Now github history of the 4 new files (`IncrementalStake.hs`, `AdaPreservation.hs`, `CollisionFreeness.hs`, `Deposits.hs`)  shows the rename which then can be followed, and blame shows like in the initial file

Reply:
Hell yeah!!! Thank you Teodora for putting this extra effort. Extra commits is a small price to pay for keeping all that git history.